ddcz/tavern.py

```diff
@@ -26,3 +32,79 @@ def get_tables_with_access(user_profile, candidate_tables_queryset):
             acls=related_permissions_map.get(table.pk, set()),
         )
     ]
+
+
+def migrate_tavern_access(
+    print_progress=True,
+    table_visitor_model=None,
+    tavern_access_model=None,
+    user_profile_model=None,
+):
+    """Migrate access privileges from putyka v0 (the DDCZ 2001 version) to putyka v1 (DDCZ 2004 version)"""
+
+    i = 0
+
+    if not table_visitor_model:
+        table_visitor_model = TavernTableVisitor
+
+    if not tavern_access_model:
+        tavern_access_model = TavernAccess
+
+    if not user_profile_model:
+        user_profile_model = UserProfile
+
+    # Completely accidentally, the `-typ_pristupu` gives us correct enough
+    # ordering of access priorities
+    for tavern_access in (
+        tavern_access_model.objects.all()
+        .select_related("id_stolu")
+        .order_by("-typ_pristupu")
+    ):
+        if print_progress and i % 100 == 0:
+            sys.stdout.write(".")
+            sys.stdout.flush()
+
+        try:
```

Comment:
I am not sure if I understand this. The previous `if` is finished here when `try` is at the same level, but at line 85 is `else`. Maybe it is the visualization here in gitlab? Or is this something I don't get?